### ISSUE TITLE
Occasionally see this when running tests

### ISSUE THREAD

**Issue by karthiknadig** [Microsoft/ptvsd#412]
```
Exception in thread ptvsd.Server:
Traceback (most recent call last):
  File "C:\GIT\ptvsd\ptvsd\ipcjson.py", line 288, in process_one_message
IndexError: pop from empty list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python37\lib\threading.py", line 917, in _bootstrap_inner
  File "C:\Python37\lib\threading.py", line 865, in run
  File "C:\GIT\ptvsd\ptvsd\wrapper.py", line 737, in process_messages
  File "C:\GIT\ptvsd\ptvsd\ipcjson.py", line 282, in process_messages
  File "C:\GIT\ptvsd\ptvsd\ipcjson.py", line 291, in process_one_message
  File "C:\GIT\ptvsd\ptvsd\ipcjson.py", line 158, in _wait_for_message
    line = self._buffered_read_line_as_ascii()
  File "C:\GIT\ptvsd\ptvsd\ipcjson.py", line 117, in _buffered_read_line_as_ascii
    temp = self.__socket.recv(1024)
OSError: [WinError 10038] An operation was attempted on something that is not a socket
```

**Comment by ericsnowcurrently** [comment]
This should be resolved now that I've landed #455.

**Comment by karthiknadig** [comment]
Will reopen if seen again.
